tsflex/features/feature_collection.py

```diff
@@ -458,6 +539,10 @@ def calculate(
                 window (which is a full) window will not be included!
                 - *(len * sampling_rate - window_size) % stride = 0*. Remark that the
                   above case is a base case of this.
+        group_by: str, optional
+            The title of the column by which to perform grouping.
+            If this parameter is used, the parameters `stride`, `segment_start_idxs`,
+            `segment_end_idxs`, `window_idx` and `include_final_window` will be ignored.
```

Comment:
What assumptions and best practices are needed when we perform such grouping based indexing?